### ISSUE TITLE
Git fails to clone Qt5

### ISSUE THREAD

**Issue by ncook-hxgn** [git-for-windows/git#3369]
I am seeing the issue as described [here](https://stackoverflow.com/questions/15607905/git-failed-when-attempting-to-retrieving-qt5-source-code-for-compiling-under-win) - that is, I am seeing a perl failure when cloning. This appears to be related to msys.dll according to the linked page.

 - [ x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup
```
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.9.5
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************

E:\repos>cmd /c ver

Microsoft Windows [Version 10.0.19043.1165]

E:\repos>git --version
git version 2.33.0.windows.1

E:\repos>git-lfs --version
git-lfs/2.13.2 (GitHub; windows amd64; go 1.14.13; git fc664697)

E:\repos>perl --version

This is perl 5, version 26, subversion 3 (v5.26.3) built for MSWin32-x64-multi-thread
(with 2 registered patches, see perl -V for more detail)

Copyright 1987-2018, Larry Wall

Binary build 0000 [6f189753] provided by ActiveState http://www.ActiveState.com
Built Oct 28 2020 16:21:55

Perl may be copied only under the terms of either the Artistic License or the
GNU General Public License, which may be found in the Perl 5 source kit.

Complete documentation for Perl, including FAQ lists, should be found on
this system using "man perl" or "perldoc perl".  If you have access to the
Internet, point your browser at http://www.perl.org/, the Perl Home Page.


E:\repos>python --version
Python 3.9.0

E:\repos>git --version --build-options
git version 2.33.0.windows.1
cpu: x86_64
built from commit: fb59c1de9fd599a3eeb8459eb7cab731c6672ea3
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon

E:\repos>
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?
```
E:\repos>type "C:\Program Files\Git\etc\install-options.txt"
Editor Option: VisualStudioCode
Custom Editor Path:
Default Branch Option:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: WinSSL
CRLF Option: CRLFAlways
Bash Terminal Option: ConHost
Git Pull Behavior Option: Merge
Use Credential Manager: Core
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled

E:\repos>
```

### Details

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```xml
<?xml version="1.0" encoding="utf-8" ?>
<Project DefaultTargets="Clone" 
  xmlns="http://schemas.microsoft.com/developer/msbuild/2003">

  <PropertyGroup>
    <SourceUrl Condition="'$(SourceUrl)' == ''">https://code.qt.io/qt/qt5.git</SourceUrl>
    <Version Condition="'$(Version)' == ''">5.12.4</Version>
    <Release Condition="'$(Release)' == ''"></Release>
    <Compiler Condition="'$(Compiler)' == ''">msvc2017</Compiler>

    <SourcePath>Q:\src\Qt-$(Version)\</SourcePath>
    <QtPath>Q:\msvc\Qt-$(Version)\</QtPath>
  </PropertyGroup>

  <Target Name="CreateWorkspace" DependsOnTargets="RemoveWorkspace">
    <Exec
      WorkingDirectory="$(MSBuildThisFileDirectory)"
      Command="subst Q: $(MSBuildThisFileDirectory)"
    />
  </Target>

  <Target Name="RemoveWorkspace">
    <Exec
      Condition="Exists('Q:\$(MSBuildThisFile)')"
      WorkingDirectory="$(MSBuildThisFileDirectory)"
      Command="subst Q: /D"
    />
  </Target>

  <!-- Clone the Git repository -->
  <Target Name="Clone" DependsOnTargets="CreateWorkspace">
    <RemoveDir
      Directories="$(SourcePath)"
    />
    <MakeDir
      Directories="$(SourcePath)"
    />
    <Exec
      WorkingDirectory="$(MSBuildThisFileDirectory)"
      Command="git clone $(SourceUrl) $(SourcePath)"
    />
    <Exec
      WorkingDirectory="$(SourcePath)"
      Command='git fetch --tags'
    />
    <Exec
      WorkingDirectory="$(SourcePath)"
      Command="git checkout $(Version)"
    />
    <Exec
      WorkingDirectory="$(SourcePath)"
      Command="perl init-repository"
    />
  </Target>
   ...
</Project>
```
 - What did you expect to occur after running these commands?

Qt 5 is cloned to Q:\

 - What actually happened instead?

```
EXEC : error : unable to create file XQTS/Queries/XQueryX/Expressions/Operators/ArithExpr/DurationArith/YearMonthDurationDivideYMD/op-divide-yearMonthDuration-by-yearMonthDuration2args-5.xqx: Filename too lon
g [E:\repos\hexagonmi\PS-DevOps\NugetPackageBuilds2\Qt\Hexagon.Qt.proj]
  fatal: Unable to checkout 'ddfcd98e8d3f78d696bb25ddcaacc3e3cba51e7f' in submodule path 'tests/auto/3rdparty/testsuites'
  git submodule update --force --no-fetch exited with status 256 at init-repository line 198.
        Qt::InitRepository::exe(Qt::InitRepository=HASH(0x21c30b0), "git", "submodule", "update", "--force", "--no-fetch") called at init-repository line 431
        Qt::InitRepository::git_clone_all_submodules(Qt::InitRepository=HASH(0x21c30b0), "qtxmlpatterns.git", 0, "/qtxmlpatterns", "all") called at init-repository line 437
        Qt::InitRepository::git_clone_all_submodules(Qt::InitRepository=HASH(0x21c30b0), "qt5", 0, "", "preview", "essential", "addon", "deprecated") called at init-repository line 645
        Qt::InitRepository::run(Qt::InitRepository=HASH(0x21c30b0)) called at init-repository line 656
E:\repos\hexagonmi\PS-DevOps\NugetPackageBuilds2\Qt\Hexagon.Qt.proj(53,5): error MSB3073: The command "perl init-repository" exited with code 2.
Done Building Project "E:\repos\hexagonmi\PS-DevOps\NugetPackageBuilds2\Qt\Hexagon.Qt.proj" (default targets) -- FAILED.


Build FAILED.
```

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

[https://code.qt.io/qt/qt5.git](https://code.qt.io/qt/qt5.git)

**Comment by ncook-hxgn** [comment]
FWIW, this MSbuild stuff is from a branch that's a few months old.. and this MSBuild Target worked, and I was compiling Qt 5.12.4. (It Used To Work And Now It Doesn't)

I suspect something has changed in Git for Windows between February 24 2021 and about a week ago when I installed Git 2.32.windows.1.. I updated to Git 2.33.windows.1 today to see if it was going to be fixed, alas.. 

Unfortunately, I cannot tell you what version of Git for Windows I had installed back in February, but it was likely to be fairly up to date at the time as my job involves updating git for windows on several machines

**Comment by ncook-hxgn** [comment]
Tomorrow I'll spend some time installing different versions of git for windows from the start of the year to help find when it broke.. because at the end of the day I need a clone of Qt5 😄

**Comment by dscho** [comment]
Maybe you could try again after setting `core.longPaths` to `true` in your global Git config? At least that _should_ fix things, according to [the "Known Issues" section of Git for Windows' Release Notes](https://github.com/git-for-windows/build-extra/blob/main/ReleaseNotes.md#known-issues):

> - Many Windows programs (including the Windows Explorer) have problems with directory trees nested so deeply that the absolute path is longer than 260 characters. Therefore, Git for Windows refuses to check out such files by default. You can overrule this default by setting `core.longPaths`, e.g. `git clone -c core.longPaths=true` ....

**Comment by ncook-hxgn** [comment]
@dscho that's a good idea - I already had to subst a drive so not a bad shout. 
`git config core.longPaths` returned nothing so I'm not sure if it's set already, however I have Enable Long Win32 Paths™ enabled in Windows Settings for various other reasons. 

I was able to work around the issue by replacing the call to `perl init-repository` with a simple `git submodule update  --init`, which I like because it's a bit less exotic. I found that I could do that whilst I was skimming over [this](https://forum.qt.io/topic/69012/queries-regarding-the-running-the-init-repository-after-downloading-the-qt-source-code/11).

Possibly it is Qt5's init-repository script that is broken (I took a look at it and it's basically shelling out to git), or at least needs a patch to be able to work with a more modern version of git or git-for-windows than it was developed against.

Either way, I should likely close this as whilst I suspect a change to the Git CLI has broken Qt's script, it is after all Qt's script to maintain. 👍
